servant/src/Servant/API/ContentTypes.hs

```diff
@@ -235,25 +243,31 @@ class (AllMime list) => AllMimeRender (list :: [*]) a where
                   -> [(M.MediaType, ByteString)]    -- content-types/response pairs
 
 instance OVERLAPPABLE_ ( MimeRender ctyp a ) => AllMimeRender '[ctyp] a where
-    allMimeRender _ a = [(contentType pctyp, mimeRender pctyp a)]
-        where pctyp = Proxy :: Proxy ctyp
+    allMimeRender _ a = [ (ct, bs) | ct <- NE.toList $ contentTypes pctyp ]
```

Comment:
Wouldn't this be easier to read without list comprehensions? Maybe that's just me.